snapshot/cli/cli.go

```diff
@@ -136,6 +142,94 @@ func (c *ingestGitCommitCommand) run(db snapshot.DB, _ *cobra.Command, _ []strin
 	return nil
 }
 
+// Subcommand for creating and uploading git bundles.
+// This is a workaround for creating arbitrary git bundles and keeping them in a Bundlestore.
+// We need this for now because generic bundles do not fit well with the existing
+// Snapshot, ID, Stream schemas.
+type createGitBundleCommand struct {
+	file  string
+	basis string
+	ref   string
+	ttld  time.Duration
+	keep  bool
+}
+
+func (c *createGitBundleCommand) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create_git_bundle",
+		Short: "Creates a git bundle file as specified by basis and reference for the cwd repo",
+	}
+	// See https://git-scm.com/docs/git-bundle for basis, ref usage
+	cmd.Flags().StringVar(&c.file, "file", "", "Output bundle file (defaults to temp file)")
+	cmd.Flags().StringVar(&c.basis, "basis", "", "Basis for bundle")
+	cmd.Flags().StringVar(&c.ref, "ref", "", "Reference to be packaged")
+	cmd.Flags().DurationVar(&c.ttld, "ttl", bundlestore.DefaultTTL, "Stored bundle TTL (duration from now)")
+	cmd.Flags().BoolVar(&c.keep, "keepFile", false, "Keep created bundles file")
+	return cmd
+}
+
+func (c *createGitBundleCommand) run(db snapshot.DB, _ *cobra.Command, _ []string) error {
+	if c.basis == "" || c.ref == "" {
+		return fmt.Errorf("Create bundle command requires a basis and reference")
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("cannot get working directory: wd")
+	}
+
+	ingestRepo, err := repo.NewRepository(wd)
+	if err != nil {
+		return fmt.Errorf("not a valid repo dir: %v, %v", wd, err)
+	}
+
+	bundleFilename, tempDir := "", ""
+	if c.file != "" {
+		bundleFilename = c.file
+	} else {
+		td, err := temp.TempDirDefault()
+		if err != nil {
+			return fmt.Errorf("Couldn't create temp dir: %v", err)
+		}
+		tempDir = td.Dir
+		bundleFilename = path.Join(tempDir, fmt.Sprintf("bs-%v-%v.bundle", c.basis, c.ref))
```

Comment:
If someone ran scoot-snapshot-db on an old commit it would create bs-<rev>.bundle which the funnel could also create, and they would collide (and be fundamentally different, I think). Is it because we don't ever expect users to be creating snapshots on existing revs?

On the bundle contents, the existing bundles created with this tool are odd; they bundle a special scoot ref "reserved_scoot/bundlestore/__temp_for_writing" and appear to have a full history from the command line although they do not. Basically the new bundles are not easily cross-compatible with our existing ones if I understand them correctly (and I may not).

Agree though that the expanded name check is ugly. If we can determine there's no danger in using the old naming let's remove it.